src/FakeItEasy/Configuration/RuleBuilder.cs

```diff
@@ -158,9 +158,7 @@ public IAfterCallSpecifiedConfiguration CallsBaseMethod()
 
             public IReturnValueConfiguration<TMember> WhenArgumentsMatch(Func<ArgumentCollection, bool> argumentsPredicate)
             {
-                Guard.AgainstNull(argumentsPredicate, nameof(argumentsPredicate));

```

Comment:
I'm doing this from web, not IDE, so may be wrong, but are we losing the Guard because `ParentConfiguration` doesn't check?